src/zenml/mlstacks/utils.py

```diff
@@ -0,0 +1,525 @@
+    stack: "Stack", components: List["Component"]
+) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """Converts mlstacks Stack and Components to dicts.
+
+    Args:
+        stack: A mlstacks Stack object.
+        components: A list of mlstacks Component objects.
+
+    Returns:
+        A tuple of Stack and List[Component] dicts.
+    """
+    verify_mlstacks_prerequisites_installation()
+
+    # convert to json first to strip out Enums objects
+    stack_dict = json.loads(stack.json())
+    components_dicts = [
+        json.loads(component.json()) for component in components
+    ]
+
+    return stack_dict, components_dicts
+
+
+def generate_unique_filename(base_filename: str) -> str:
+    """
+    Generates a unique filename by appending a UUID to the base filename.
+
+    Args:
+        base_filename: The base filename.
+
+    Returns:
+        The unique filename.
+    """
+    unique_suffix = (
+        uuid.uuid4().hex
+    )  # Generate a random UUID and convert it to a string.
+    return f"{base_filename}_{unique_suffix}"
+
+
+@verify_installation
+def import_new_mlstacks_stack(
+    stack_name: str, provider: str, stack_spec_dir: str
+) -> None:
+    """Import a new stack deployed for a particular cloud provider.
+
+    Args:
+        stack_name: The name of the stack to import.
+        provider: The cloud provider for which the stack is deployed.
+        stack_spec_dir: The path to the directory containing the stack spec.
+    """
+    from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+    from mlstacks.utils import terraform_utils
+
+    tf_dir = f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/terraform/{provider}-modular"
+    stack_spec_file = f"{stack_spec_dir}/stack-{stack_name}.yaml"
+    # strip out the `./` from the stack_file_path
+    stack_filename = terraform_utils.get_stack_outputs(
+        stack_spec_file, output_key="stack-yaml-path"
+    ).get("stack-yaml-path")[2:]
+    import_stack_path = f"{tf_dir}/{stack_filename}"
+    data = read_yaml(import_stack_path)
+    # import stack components
+    component_ids = {}
+    for component_type_str, component_config in data["components"].items():
+        component_type = StackComponentType(component_type_str)
+        component_spec_path = f"{stack_spec_dir}/{component_config['flavor']}-{component_type_str}.yaml"
+
+        from zenml.cli.stack import _import_stack_component
+
+        component_id = _import_stack_component(
+            component_type=component_type,
+            component_dict=component_config,
+            component_spec_path=component_spec_path,
+        )
+        component_ids[component_type] = component_id
+
+    imported_stack = Client().create_stack(
+        name=stack_name,
+        components=component_ids,
+        is_shared=False,
+        stack_spec_file=stack_spec_file,
+    )
+
+    print_model_url(get_stack_url(imported_stack))
+
+
+@verify_installation
+def import_new_mlstacks_component(
```

Comment:
this has a lot of duplicate code with `import_new_mlstacks_stack`

Reply:
I have now refactored to remove the duplication.